README.md

```diff
@@ -258,7 +258,28 @@ SuperLogin supports two types of workflows for OAuth2 providers: popup window an
 
 #### Popup Window Workflow for web browsers (desktop and mobile)
 
-Your client must create a popup window and point it to `/{provider}`, where the user will be directed to authenticate with that provider. After authentication succeeds or fails, it will call a Javascript callback on the parent window called `superlogin.oauthSession`.
+Your client must create a popup window and point it to `/{provider}`, where the user will be directed to authenticate with that provider. After authentication, succeeds or fails, it will [post a message](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to the parent window with the data set to `{error, session, link }`.
+
```

Comment:
Thank for fixing, looks good to me. 

Would you add a little warning to be sure to override the`*` - origin in `auth-callback.ejs` when actually using this in production?

Reply:
Yeah I can add it, but I don't know if the warning should say that they need to overwrite the `auth-callback.ejs` inside `node_modules/@sl-nx/superlogin-next/templates/oauth/` or I could add a property `oauth` in the config that has a `targetOrigin`  property, e.g.
```
export interface OAuthConfig {
  /**
  * Always provide a specific [targetOrigin](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#syntax), not `*`, if you know where
  * the other window's document should be located. Failing to provide
  * a specific target discloses the data you send to any interested 
  * malicious site. 
  */
  targetOrigin?: string; 
}
```
Then the oauth config could be passed to the render template
```
// oauth.ts line 50 & 97
const html = render(template, {...results, ...this.config.oauth});
```
https://github.com/sl-nx/superlogin-next/blob/14399bbc7a5ce6abd70f7d4f04ae39c825bcd024/src/oauth.ts#L37-L51
And finally modify the `auth-callback.ejs`
```
  var targetOrigin = <%- JSON.stringify(targetOrigin || '*') %>;
  window.opener.postMessage({error, session, link}, targetOrigin);
```
Or maybe there is some way to pass the template path that I'm not aware of.
Any thoughts or comments are really appreciated

Reply:
Yeah, the way it is currently implemented no custom template will be loaded. If we'd like to make it more flexible, it should follow the same logic as for the email templates. Do you think that there is a use case to make more modifications in that template than just the `targetOrigin`? I guess yes, so the whole template should be customisable, else just the origin as in your suggestion would be suitable.

I'd say:
- add the corresponding template entry to the `ProviderConfig` (l. 269 in `config.d.ts`)  for each provider, assuming that the options might be different for each of them.
- as you suggested, use the custom template in `linkSuccess`, `initSession` and `oauthErrorHandler`, falling back to the default if none is given.
- looking at the code, a small refactoring to retrieve the `template` with a single method would be nice.

Reply:
Ok I think I have done it, let me know what you think.